mentii/mentii
# Problem bad step

This pull request takes a solution path that is generated from mathsteps, modifies steps to create a "bad" step, and then generates bad step paths from then on.

**Significant Changes**
1. Created an algebra.py file to handle generation of bad steps
2. Implemented a function `generateBadSteps()` to randomly pick failure points, modify the step, and generate a path from that modified step
3. Implemented a function `modifySteps()` that reverses operators to create "bad" steps
4. Unit test for `generateBadSteps()` in `test_algebra.py`

**How to Test**
1. navigate to test directory in Backend
2. run `python test_algebra.py`
3. make sure all test cases pass

**Notes**
:clap:
